Code/GraphMol/MarvinParse/MarvinDefs.cpp

```diff
@@ -0,0 +1,2978 @@
+    }
+
+    orphanedBondToFix->atomRefs2[1] =
+        orphanedBonds[matchedOrphanBondIndex]
+            ->atomRefs2[0];  // [0] is the orpahened atom (still in the mol)
+
+    // undelete the bond which has been fixed (really remove it from the list
+    // of bonds to be delete)
+
+    bondsToDelete.erase(
+        find(bondsToDelete.begin(), bondsToDelete.end(), orphanedBondToFix));
+
+    // any siblings or children that reference the matched Orphaned bond must
+    // be fixed to reference the retained (fixed) orphan bond
+
+    for (auto siblingSgroup :
+         this->parent->sgroups)  // note: includes this sgroups, which is
+                                 // technically NOT a sibling
+    {
+      for (auto &bond : siblingSgroup->bonds) {
+        if (bond->id == orphanedBonds[matchedOrphanBondIndex]->id) {
+          bond = orphanedBondToFix;
+        }
+      }
+    }
+
+    // it COULD happen that children referenced the matched orphan bond
+
+    for (auto childSgroup : this->sgroups) {
+      for (auto childBond : childSgroup->bonds) {
+        if (childBond->id == orphanedBonds[matchedOrphanBondIndex]->id) {
+          childBond = orphanedBondToFix;
+        }
+      }
+    }
+
+    orphanedBonds.erase(orphanedBonds.begin() + matchedOrphanBondIndex);
+  };
+
+  // if we get here, all the changes to be made have been determined. so
+  // make them
+
+  for (auto childSgroup : sgroupsToDelete) {
+    sgroups.erase(std::find(sgroups.begin(), sgroups.end(), childSgroup));
+    delete childSgroup;
+  }
+
+  // remove the atoms
+
+  for (auto atomPtr : atomsToDelete) {
+    for (auto thisParent = this->parent;;
+         thisParent = thisParent->parent) {  // do all parents and grandparents
+                                             // ... to to the actual parent
+
+      thisParent->atoms.erase(std::find(thisParent->atoms.begin(),
+                                        thisParent->atoms.end(), atomPtr));
+
+      if (thisParent == actualParent) {
+        break;
+      }
+    }
+
+    // also delete the atoms from this multiple group AND any sibling sgroups
+
+    for (auto siblingSgroup : this->parent->sgroups) {
+      auto siblingAtomPtr = std::find(siblingSgroup->atoms.begin(),
+                                      siblingSgroup->atoms.end(), atomPtr);
+      if (siblingAtomPtr != siblingSgroup->atoms.end()) {
+        siblingSgroup->atoms.erase(siblingAtomPtr);
+      }
+    }
+
+    delete atomPtr;
+  }
+
+  atomsToDelete.clear();
+
+  // remove the bonds
+
+  for (MarvinBond *bondPtr : bondsToDelete) {
+    actualParent->bonds.erase(std::find(actualParent->bonds.begin(),
+                                        actualParent->bonds.end(), bondPtr));
+    delete bondPtr;
+  }
+  bondsToDelete.clear();
+
+  this->isExpanded = false;
+}
+
+IsSgroupInAtomSetResult MarvinMolBase::isSgroupInSetOfAtoms(
+    std::vector<MarvinAtom *> &setOfAtoms) const {
```

Comment:
I think this could be:
```suggestion
IsSgroupInAtomSetResult MarvinMolBase::isSgroupInSetOfAtoms(
    const std::vector<MarvinAtom *> &setOfAtoms) const {
```

Reply:
yes - it is changed